### PR TITLE
fix(comments): pagination of comments continues to the comment section

### LINKED ISSUE #5041: Improve comment continuation (Trac #5041)
_Original ticket http://trac.elgg.org/ticket/5041 on 43103742-02-22 by trac user mrclay, assigned to unknown._

Elgg version: 1.8

E.g. object has long description and 40 comments. You scroll through description and read first 25 comments, hit next. Now you have to scroll through the entire description again, and worse, there's no indication that the first comment you see is actually the 26th.

I suggest continuation pages (as long as we must have them) should only have an excerpt/search listing view of the object (linked to the first page) and an additional pagination control set on top.

Short term: add id="elgg-comments" to comments listing and have comment pagination links include that fragment. At least minimize discussion continuations as first step since that will benefit community.

### PR DESCRIPTION
**Pull request by jeabakker**
When clicking on the next/previous links of the comments, you would end
up on the top of the page, it would be nice if you could directly
continue reading the next set of comments

fixes #5041

### CODE REVIEW

File: views/default/page/elements/comments.php [hunk 1]
```diff
@@ -19,22 +19,25 @@
 	$limit = elgg_trigger_plugin_hook('config', 'comments_per_page', [], 25);
 }
 
-$id = '';
-if (isset($vars['id'])) {
-	$id = "id=\"{$vars['id']}\"";
-}
+$attr = [
+	'id' => elgg_extract('id', $vars, 'elgg-comments')
```
Comment (mrclay) [inline views/default/page/elements/comments.php]: This assumes 1 use of this view per page. That's probably a safe assumption.

File: views/default/page/elements/comments.php [hunk 2]
```diff
@@ -19,22 +19,25 @@
...
+];
 
 $class = 'elgg-comments';
 if (isset($vars['class'])) {
 	$class = "$class {$vars['class']}";
 }
+$attr['class'] = $class;
```
Comment (mrclay) [inline views/default/page/elements/comments.php]: @jdalsem has an elegant refactoring. Replace this and the previous 4 lines with:
```php
$attr['class'] = (array) elgg_extract('class', $vars, []);
$attr['class'][] = 'elgg-comments';
```

File: views/default/page/elements/comments.php [hunk 3]
```diff
@@ -19,22 +19,25 @@
...
 
 // work around for deprecation code in elgg_view()
 unset($vars['internalid']);
 
-echo "<div $id class=\"$class\">";
+// make the pagination go to the comments section
```
Comment (mrclay) [inline views/default/page/elements/comments.php]: Can we not do this when going to the first page?

**Comment by jeabakker** [comment]
refactored to @jdalsem class method

**Comment by mrclay** [comment]
Could this be more simply done by adding the fragments to the URLs in JS?

**Comment by jeabakker** [comment]
that would require a lot of js, plus then it wont work if js is disabled

**Comment by mrclay** [comment]
JS: find pagination URLs inside .elgg-comments. For those href containing offset, add fragment. Seems like a few lines, but the semantic argument does appeal to me (PHP should spit out complete URLs).

@hypeJunction @ewinslow any thoughts here?

File: views/default/page/elements/comments.php (hunk 3 above)
Comment (mrclay) [inline views/default/page/elements/comments.php]: I feel like this functionality belongs in the pagination view. Setting base_url here feels a little hacky.

Comment (mrclay) [inline views/default/page/elements/comments.php]: I think setting base_url will break plugins fetching from XHR. Maybe only set it if `!elgg_is_xhr()`

**Comment by mrclay** [comment]
I put some refactoring into this: https://github.com/jeabakker/Elgg/pull/1 which would allow easily adding this functionality to any list with a known ID.

**Comment by hypeJunction** [comment]
I have moved away from using IDs for anything that contains a GUID. There are always cases when those are not unique on the page. As such, I wouldn't use URL fragments, and lookup DOM parent instead.

**Comment by mrclay** [comment]
@hypeJunction so how should we add this functionality in core?

**Comment by hypeJunction** [comment]
I'd almost say let's ajaxify comments pagination, but I guess that doesn't doesn't solve the problem at hand here. I am fine with this PR but it should at least try and generate a unique id (add an iterated suffix perhaps). Alternatively we could use jquery ui uniqueId and rewrite URL fragments client side

**Comment by mrclay** [comment]
More expanded version #8226

**Comment by mrclay** [comment]
Merged #8226

**Pull request closed without merging**